plugins/polio/api/vaccines/repository.py

```diff
@@ -0,0 +1,260 @@
+    form_a_data = serializers.SerializerMethodField()
+
+    def get_vrf_data(self, obj):
+        vrfs = VaccineRequestForm.objects.filter(campaign=obj.campaign)
+        return [
+            {
+                "date": vrf.date_vrf_reception,
+                "file": vrf.document.url if vrf.document else None,
+                "is_missing": vrf.vrf_type == VaccineRequestFormType.MISSING,
+                "is_not_required": vrf.vrf_type == VaccineRequestFormType.NOT_REQUIRED,
+            }
+            for vrf in vrfs
+        ]
+
+    def get_pre_alert_data(self, obj):
+        pre_alerts = VaccinePreAlert.objects.filter(request_form__campaign=obj.campaign)
+        return [
+            {"date": pa.date_pre_alert_reception, "file": pa.document.url if pa.document else None} for pa in pre_alerts
+        ]
+
+    def get_form_a_data(self, obj):
+        form_as = OutgoingStockMovement.objects.filter(campaign=obj.campaign, round=obj)
+        return [
+            {
+                "date": fa.form_a_reception_date,
+                "file": fa.document.url if fa.document else None,
+                "is_late": fa.form_a_reception_date > (obj.ended_at + timedelta(days=14))
+                if fa.form_a_reception_date and obj.ended_at
+                else None,
+            }
+            for fa in form_as
+        ]
+
+
+class VaccineReportingFilterBackend(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        # Filter by campaign status
+        campaign_status = request.query_params.get("campaign_status", None)
+
+        # Get campaign dates subquery
+        campaign_dates = (
+            Round.objects.filter(campaign=OuterRef("campaign"))
+            .values("campaign")
+            .annotate(campaign_started_at=Min("started_at"), campaign_ended_at=Max("ended_at"))
+        )
+
+        # Add campaign dates to main queryset
+        queryset = queryset.annotate(
+            campaign_started_at=Subquery(campaign_dates.values("campaign_started_at")),
+            campaign_ended_at=Subquery(campaign_dates.values("campaign_ended_at")),
+        )
+
+        if campaign_status:
+            today = datetime.now().date()
+            if campaign_status.upper() == "ONGOING":
+                queryset = queryset.filter(campaign_started_at__lte=today, campaign_ended_at__gte=today)
+            elif campaign_status.upper() == "PAST":
+                queryset = queryset.filter(campaign_started_at__lt=today, campaign_ended_at__lt=today)
+            elif campaign_status.upper() == "PREPARING":
+                queryset = queryset.filter(campaign_started_at__gte=today)
+
+        # Filter by country block
+        country_block = request.query_params.get("country_block", None)
+        if country_block:
+            queryset = queryset.filter(campaign__country__groups__in=country_block.split(","))
+
+        # Filter by country (multi)
+        countries = request.query_params.get("countries", None)
```

Comment:
Relying on `request.query_params` is prone to errors because there is no validation of the data.

E.g. this is how to break the endpoint: `http://localhost:8081/api/polio/vaccine/repository/?country_block=foo,bar`.

```
ValueError at /api/polio/vaccine/repository/
Field 'id' expected a number but got 'foo'.
```

Using a [`FilterSet`](https://django-filter.readthedocs.io/en/latest/guide/rest_framework.html#adding-a-filterset-with-filterset-class) is a good way to go to avoid such pitfalls because it comes with built-in validation.

As a quick fix, maybe you can just validate that values are `int`s.